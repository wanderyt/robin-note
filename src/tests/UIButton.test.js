import React from 'react';
// import jest from 'jest';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import UIButton from '../scripts/uikit/button';

describe('UIButton', () => {
  it('should render correctly', () => {
    const output = shallow(
      <UIButton
        text='my button'
        handleClick={() => alert('clicked!')} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should handle Click event', () => {
    console.log = jest.fn();
    const output = shallow(
      <UIButton
        text='my button'
        handleClick={() => console.log('clicked!')} />
    );
    output.simulate('click');
    expect(console.log).toHaveBeenCalledWith('clicked!');
  });
});