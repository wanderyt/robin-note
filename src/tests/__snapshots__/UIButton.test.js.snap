// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIButton should render correctly 1`] = `
<button
  className="UIButton "
  onClick={[Function]}
>
  my button
</button>
`;
